### PR TITLE
confine pulpcore cli checks for setups where the cli is available

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: test/definitions/checks/pulpcore/no_running_tasks_test.rb [hunk 1]
```diff
@@ -8,14 +8,26 @@
       Checks::Pulpcore::NoRunningTasks.new
     end
 
+    it 'passes when no cli is installed' do
+      assume_feature_present(:pulpcore, :running_tasks => []) do |feature|
+        feature.any_instance.stubs(:cli_available?).returns(false)
+      end
+      result = run_check(subject)
+      assert result.success?, 'Check expected to succeed'
```
Comment (evgeni) [inline test/definitions/checks/pulpcore/no_running_tasks_test.rb:16]: interestingly, checks that do not pass confinement are still reported as `success`, not `skipped` :woman_shrugging:

**Comment by evgeni** [comment]
Verified by

```
% ansible-playbook pipelines/install_pipeline.yml -e pipeline_os=centos9-stream -e pipeline_type=katello -e pipeline_version=nightly -e '{"packit_prs":["https://github.com/theforeman/foreman_maintain/pull/921"]}'
```

File: definitions/features/pulpcore.rb [hunk 2]
```diff
@@ -15,6 +16,10 @@ class Features::Pulpcore < ForemanMaintain::Feature
     end
   end
 
+  def cli_available?
```
Comment (evgeni) [inline definitions/features/pulpcore.rb:19]: I guess, design wise, this should be moved to an own feature, but I didn't want to refactor all of this in a bugfix PR.

**Review by ekohl — APPROVED** [review]

**Comment by lpramuk** [comment]
Successfully tested. 
Applied patch downstream and Capsule 6.15.3 upgraded to 6.16.0 Snap1 successfully

**Pull request merged**